main.go

```diff
@@ -121,25 +106,49 @@ func main() {
 		})
 	}
 
-	stopOplogTail := make(chan bool)
-	waitGroup.Add(1)
-	go func() {
-		tailer := oplog.Tailer{
-			MongoClient:  mongoSession,
-			RedisClients: aggregatedRedisClients[0], // the tailer coroutine needs a redis client for determining start timestamp
-			// it doesn't really matter which one since this isn't a meaningful amount of load, so just take the first one
-			RedisPrefix: config.RedisMetadataPrefix(),
-			MaxCatchUp:  config.MaxCatchUp(),
-			Denylist:    &denylist,
+	readParallelism := config.ReadParallelism()
+
+	stopOplogTails := make([]chan bool, readParallelism)
+	aggregatedMongoSessions := make([]*mongo.Client, readParallelism)
+	for i := 0; i < readParallelism; i++ {
+		mongoSession, err := createMongoClient()
+		if err != nil {
+			panic(fmt.Sprintf("[%d] Error initializing oplog tailer: %s", i, err.Error()))
 		}
-		tailer.Tail(aggregatedRedisPubs, stopOplogTail)
+		defer func(i int) {
+			mongoCloseCtx, cancel := context.WithTimeout(context.Background(), config.MongoConnectTimeout())
```

Comment:
Is this right? It's applying the "connect" timeout the the "Disconnect" operation?

Reply:
This is not new, OTR already did this (for the singleton read). It's a little iffy to use the same timeout for connect/disconnect but it hasn't caused problems historically, i'm inclined to keep it.

Reply:
yeah, i think not a major concern, just confirming that it wasn't a goof